icu4c/source/data/misc/timezoneTypes.txt

```diff
@@ -5,14 +5,47 @@ timezoneTypes:table(nofallback){
     bcpTypeAlias{
         tz{
             aqams{"nzakl"}
+            aukns{"auhba"}
+            caffs{"cawnp"}
             camtr{"cator"}
+            canpg{"cator"}
+            capnt{"caiql"}
+            cathu{"cator"}
+            cayzf{"caedm"}
             cnckg{"cnsha"}
             cnhrb{"cnsha"}
             cnkhg{"cnurc"}
             gaza{"gazastrp"}
+            mxstis{"mxtij"}
+            uaozh{"uaiev"}
+            uauzh{"uaiev"}
+            umjon{"ushnl"}
             usnavajo{"usden"}
         }
     }
+    ianaMap{
+        timezone{
+            "Africa:Asmera"{"Africa/Asmara"}
+            "America:Buenos_Aires"{"America/Argentina/Buenos_Aires"}
+            "America:Catamarca"{"America/Argentina/Catamarca"}
+            "America:Coral_Harbour"{"America/Atikokan"}
+            "America:Cordoba"{"America/Argentina/Cordoba"}
+            "America:Godthab"{"America/Nuuk"}
+            "America:Indianapolis"{"America/Indiana/Indianapolis"}
+            "America:Jujuy"{"America/Argentina/Jujuy"}
+            "America:Louisville"{"America/Kentucky/Louisville"}
+            "America:Mendoza"{"America/Argentina/Mendoza"}
+            "Asia:Calcutta"{"Asia/Kolkata"}
+            "Asia:Katmandu"{"Asia/Kathmandu"}
+            "Asia:Rangoon"{"Asia/Yangon"}
+            "Asia:Saigon"{"Asia/Ho_Chi_Minh"}
+            "Atlantic:Faeroe"{"Atlantic/Faroe"}
+            "Europe:Kiev"{"Europe/Kyiv"}
+            "Pacific:Enderbury"{"Pacific/Kanton"}
+            "Pacific:Ponape"{"Pacific/Pohnpei"}
+            "Pacific:Truk"{"Pacific/Chuuk"}
+        }
+    }
```

Comment:
I see in the code what this new table is doing, but it wasn't clear to me from the rest of this PR what all the other changes to this file are doing or how they relate to the new API you're adding.  Maybe this is obvious to everybody who's spent a lot of time in the time-zone stuff, but would you mind explaining for the benefit of people like me who haven't?

Reply:
Whoops.  Didn't see the "Incomplete" label until now.  Maybe that explains my questions...

Reply:
@richgillam 
>but it wasn't clear to me from the rest of this PR what all the other changes to this file are doing or how they relate to the new API you're adding.

To support the new data structure and produce a new table in ICU data files, we need to update cldr2icu tool. These changes found under tools were made for the data mapping. All other files in icu4c/icu4j are all related to 1) new data .txt 2) API and implementation 3) test cases.

Do you see something unrelated to this work?

Reply:
My question was specifically about `timeZoneTypes.txt` and the changes to resources in it other than the new `ianaMap` resource.  Are the other changes in there somehow necessary to make the new `getIanaID()` functions work, or were they just other changes to the time-zone data in CLDR that came "along for the ride" when you re-ran the CLDR-to-ICU tools?  If they're part of the implementation of `getIanaID()`, I couldn't figure out what they were doing.

Reply:
OK. There are a few goals in CLDR-14453. First, sort out what set is appropriate for CLDR canonical zone IDs. And the second part is to provide mapping from CLDR canonical zone ID to IANA preferred zone ID. On CLDR side, we come up with new policy for a set of CLDR canonical zones. This was required to define IANA preferred zone ID mapping. This is the reason why you see a bunch of changes in timeZoneTypes.txt.

Reply:
That makes sense.  Thank you for the explanation!  I guess those changes either don't result in a caller-visible change in behavior, or CLDR has judged any changes in behavior to be beneficial?

Reply:
>I guess those changes either don't result in a caller-visible change in behavior, or CLDR has judged any changes in behavior to be beneficial?

Let me use an actual example. America/Thunder_Bay and America/Toronto. Previously, these zones are defined differently.
```
Zone America/Toronto	-5:17:32 -	LMT	1895
			-5:00	Canada	E%sT	1919
			-5:00	Toronto	E%sT	1942 Feb  9  2:00s
			-5:00	Canada	E%sT	1946
			-5:00	Toronto	E%sT	1974
			-5:00	Canada	E%sT
Zone America/Thunder_Bay -5:57:00 -	LMT	1895
			-6:00	-	CST	1910
			-5:00	-	EST	1942
			-5:00	Canada	E%sT	1970
			-5:00	Toronto	E%sT	1973
			-5:00	-	EST	1974
			-5:00	Canada	E%sT
```
Above definition is found in 2014g northamerica file. Both were in zone.tab, and some implementations show these two zones in the zone selection list. CLDR treated both of them as canonical zones. Since then, time zone database maintainer decided to make America/Thunder_Bay as "Link" of America/Toronto in backward file as below.
```
Link	America/Toronto		America/Thunder_Bay
```
The maintainer pointed out the difference before 1974 were not coming from reliable source, and concluded not worth keeping America/Thunder_Bay from America/Toronto. So he decided to move America/Thunder_Bay to backward file and use the rule same with America/Toronto.

ICU still support both America/Toronto and America/Thunder_Bay to create a time zone instance. Previously, getCanonicalID("America/Thunder_Bay") returns "America/Thunder_Bay, but with this change, it will be "America/Toronto".

In CLDR, there are two ways to define display names. One is through meta zone. Toronto and New_York share same zone display name like - "Eastern Time". In CLDR, we maintain the display name associated with meta zone America_Eastern, and maintain historic mapping between each canonical zone to meta zone. Another way to define display name is to assign names directly to canonical zone ID. The latter override the meta zone based display names.

So, deprecating America/Thunder_Bay as canonical zone and make it an alias of America/Toronto may change display names if there are zone based display name overrides (not by metazone and mapping), then display names will be changed. But assigning display names directly with zone ID is rare exception in CLDR at this moment. So practically, it should have no impacts.

Only possible change is - location based display name. In CLDR/ICU, date format patterm VVV shows exemplar city name. To produce a display name for VVV, CLDR maintains exemplar city translation. For example, "サンダーベイ" for America/Thunder_Bay in Japanese. To resolve the name, ICU gets a zone's canonical ID first, then look up the time zone exemplar city data. Previously it was America/Thunder_Bay, but it will be changed to America/Toronto. This is the actual impact.

From the time zone database maintainer's point of view, America/Thuder_Bay is only kept for backward compatibility purpose (therefore, it's defined in backward file). This means, time zone database consumer should not include America/Thunder_Bay in the zone selection list. For example, you can no longer select Thunder Bay as the system time zone in the current version of Ubuntu. VVV for America/Thunder_Bay produces "トロント" (Toronto) instead of "サンダーベイ" might not be great, but user is not supposed to select America/Thunder_Bay as system time zone any more.

Reply:
I see.  Thanks for the long explanation!  This makes sense, and I'm personally comfortable with what you're describing.